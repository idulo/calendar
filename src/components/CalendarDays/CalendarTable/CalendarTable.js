import React from "react";

import STYLES from "./CalendarTable.module.css";

const CalendarCell = ({ className, day }) => (
  <td className={className}>{day}</td>
);

const CalendarRow = ({ cells }) => <tr>{cells}</tr>;

const CalendarTable = ({ daysInMonth, firstDayOfMonth }) => {
  let calendarRow = [];
  let calendarDays = [];
  let day = 1;

  if (firstDayOfMonth > 0) {
    for (let emptyDay = 1; emptyDay <= firstDayOfMonth; emptyDay++) {
      calendarRow.push(<CalendarCell className={STYLES.emptyCell} />);
    }
  }

  while (day <= daysInMonth) {
    if (calendarRow.length === 7) {
      calendarDays.push(
        <CalendarRow
          rowCount={`row-${calendarDays.length + 1}`}
          cells={calendarRow}
        />
      );
      calendarRow = [];
    } else {
      calendarRow.push(
        <CalendarCell
          className={STYLES.dayCell}
          key={`cell-${day}`}
          day={day}
        />
      );
      day++;
    }
  }

  if (calendarRow.length > 0) {
    calendarDays.push(
      <CalendarRow key={`row-${calendarDays.length + 1}`} cells={calendarRow} />
    );
  }

  return calendarDays;
};

export default CalendarTable;
