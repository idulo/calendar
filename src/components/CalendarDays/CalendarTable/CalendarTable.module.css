.emptyCell {
  background-color: #dfdfdf;
  border-radius: 10px;
}

.dayCell {
  background-color: #e9eb87;
  border-radius: 10px;
}
