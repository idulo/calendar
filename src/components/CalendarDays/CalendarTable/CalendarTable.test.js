import React from "react";
import ReactDOM from "react-dom";
import { create } from "react-test-renderer";
import CalendarTable from "./CalendarTable";

describe("CalendarTable", () => {
  const toggleReminder = jest.fn();

  it("renders without crashing", () => {
    const div = document.createElement("div");

    ReactDOM.render(
      <CalendarTable
        toggleReminder={toggleReminder}
        daysInMonth={30}
        firstDayOfMonth={2}
      />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders correctly with all days of week", () => {
    const Week = create(
      <CalendarTable
        toggleReminder={toggleReminder}
        daysInMonth={30}
        firstDayOfMonth={2}
      />
    );

    expect(Week.toJSON()).toMatchSnapshot();
  });
});
