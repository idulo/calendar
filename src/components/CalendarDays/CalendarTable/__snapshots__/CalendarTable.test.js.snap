// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarTable renders correctly with all days of week 1`] = `
Array [
  <tr>
    <td
      className="emptyCell"
    />
    <td
      className="emptyCell"
    />
    <td
      className="dayCell"
      onClick={[Function]}
    >
      1
    </td>
    <td
      className="dayCell"
      onClick={[Function]}
    >
      2
    </td>
    <td
      className="dayCell"
      onClick={[Function]}
    >
      3
    </td>
    <td
      className="dayCell"
      onClick={[Function]}
    >
      4
    </td>
    <td
      className="dayCell"
      onClick={[Function]}
    >
      5
    </td>
  </tr>,
  <tr>
    <td
      className="dayCell"
      onClick={[Function]}
    >
      6
    </td>
    <td
      className="dayCell"
      onClick={[Function]}
    >
      7
    </td>
    <td
      className="dayCell"
      onClick={[Function]}
    >
      8
    </td>
    <td
      className="dayCell"
      onClick={[Function]}
    >
      9
    </td>
    <td
      className="dayCell"
      onClick={[Function]}
    >
      10
    </td>
    <td
      className="dayCell"
      onClick={[Function]}
    >
      11
    </td>
    <td
      className="dayCell"
      onClick={[Function]}
    >
      12
    </td>
  </tr>,
  <tr>
    <td
      className="dayCell"
      onClick={[Function]}
    >
      13
    </td>
    <td
      className="dayCell"
      onClick={[Function]}
    >
      14
    </td>
    <td
      className="dayCell"
      onClick={[Function]}
    >
      15
    </td>
    <td
      className="dayCell"
      onClick={[Function]}
    >
      16
    </td>
    <td
      className="dayCell"
      onClick={[Function]}
    >
      17
    </td>
    <td
      className="dayCell"
      onClick={[Function]}
    >
      18
    </td>
    <td
      className="dayCell"
      onClick={[Function]}
    >
      19
    </td>
  </tr>,
  <tr>
    <td
      className="dayCell"
      onClick={[Function]}
    >
      20
    </td>
    <td
      className="dayCell"
      onClick={[Function]}
    >
      21
    </td>
    <td
      className="dayCell"
      onClick={[Function]}
    >
      22
    </td>
    <td
      className="dayCell"
      onClick={[Function]}
    >
      23
    </td>
    <td
      className="dayCell"
      onClick={[Function]}
    >
      24
    </td>
    <td
      className="dayCell"
      onClick={[Function]}
    >
      25
    </td>
    <td
      className="dayCell"
      onClick={[Function]}
    >
      26
    </td>
  </tr>,
  <tr>
    <td
      className="dayCell"
      onClick={[Function]}
    >
      27
    </td>
    <td
      className="dayCell"
      onClick={[Function]}
    >
      28
    </td>
    <td
      className="dayCell"
      onClick={[Function]}
    >
      29
    </td>
    <td
      className="dayCell"
      onClick={[Function]}
    >
      30
    </td>
  </tr>,
]
`;
